Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -507,6 +520,74 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::slice)
     return js_string(vm, string_part);
 }
 
+JS_DEFINE_NATIVE_FUNCTION(StringPrototype::split)
+{
+    // FIXME Implement the @@split part
+
+    auto string = ak_string_from(vm, global_object);
+    if (string.is_null())
+        return {};
+
+    auto* result = Array::create(global_object);
+    size_t result_len = 0;
+
+    auto limit = static_cast<u32>(MAX_U32);
+    if (!vm.argument(1).is_undefined()) {
+        limit = vm.argument(1).to_u32(global_object);
+        if (vm.exception())
+            return {};
+    }
+
+    auto separator = vm.argument(0).to_string(global_object);
+    if (vm.exception())
+        return {};
+
+    if (limit == 0)
+        return result;
+
+    if (vm.argument(0).is_undefined()) {
+        result->define_property(0, js_string(vm, string));
+        return result;
+    }
+
+    auto len = string.length();
+    auto separator_len = separator.length();
+    if (len == 0) {
+        if (separator_len > 0)
+            result->define_property(0, js_string(vm, string));
+        return result;
+    }
+
+    size_t start = 0;
+    auto pos = start;
+    if (separator_len == 0) {
+        for (pos = 0; pos < len; pos++)
+            result->define_property(pos, js_string(vm, string.substring(pos, 1)));
+        return result;
+    }
+
+    while (pos != len) {
+        auto e = split_match(string, pos, separator);
```

Comment:
Any reason to avoid `memmem` here?

Reply:
I wanted to explicitly define this operation in terms of what the standard talks about. It uses AK's `starts_with()` under the hood, and that uses `memcmp` I think.

Reply:
I meant the "check equality for every character in string" part, `memmem` would probably be a fair bit faster than this for small strings, and it would give you the same-ish interface as `split_match` too.

if it's to follow the spec though, :shrug:

Reply:
I just translated the pseudo code to C++ without getting smart about it. I think that there's nothing bad in following the spec to the letter even if it leads to sub-optimal performance at this stage. We can always iterate later and have the naive-yet-standard implementation as the baseline.

Reply:
Following the spec is a fairly valid reason, I suppose.